### PR TITLE
kodi.packages.controller-topology-project: init at unstable-2022-01-22

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
`kodi-game/kodi-game-controllers` was been replaced with `kodi-game/controller-topology-project` quite some time ago.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/kodi-packages.nix [hunk 1]
```diff
@@ -171,4 +153,8 @@ let self = rec {
   trakt-module = callPackage ../applications/video/kodi/addons/trakt-module { };
 
   trakt = callPackage ../applications/video/kodi/addons/trakt { };
+
+  # deprecated or renamed packages
+
+  controllers = throw "kodi.packages.controllers has been replaced with kodi.packages.controller-topology-project - a package which contains a large number of controller profiles." { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/kodi-packages.nix:159]: That should be gated behind config.aliases.

Comment (aanderse) [reply pkgs/top-level/kodi-packages.nix:159]: Do you mean `aliases.nix`? If so, I feel like this scenario is the same as our discussion [here](https://github.com/NixOS/nixpkgs/pull/117269#discussion_r599035419) - `aliases.nix` doesn't work for packages inside things like `kodi.packages`...

Comment (SuperSandro2000) [reply pkgs/top-level/kodi-packages.nix:159]: I imagined to do it like this https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome/default.nix#L283

Comment (aanderse) [reply pkgs/top-level/kodi-packages.nix:159]: Great! Thanks for the feedback, I'll get that sorted today.

Comment (aanderse) [reply pkgs/top-level/kodi-packages.nix:159]: Let me know if this works for you @SuperSandro2000. Thanks!

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 155249` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>kodiPackages.controller-topology-project</li>
</ul>
</details>

**Comment by aanderse** [comment]
As always, thank you very much @SuperSandro2000!

**Pull request merged**